pipelines/Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+import static groovy.json.JsonOutput.toJson
+import static groovy.json.JsonOutput.prettyPrint
+
+@NonCPS
+def static parseEvictableHeroArtifacts(String fetchResponse, Integer max) {
+    def dateFmt = "yyyy-MM-dd'T'HH:mm:ss.SSS'+'"
+    def slurper = new groovy.json.JsonSlurper()
+
+    def artifacts = slurper.parseText(fetchResponse).items;
+    def records = artifacts.collect { [
+        repo: it.name,
+        pull: it.version,
+        lastModified: Date.parse(dateFmt, it.assets?.find().lastModified),
+        downloadUrl: it.assets?.find().downloadUrl
+    ] }
+
+    records.sort{ it.lastModified }
+    def recordsCount = records.size()
+    if (recordsCount <= max) {
+        return []
+    }
+    // Note: Groovy subscript operator is INCLUSIVE on both sides, so offset by 1
+    def lastToEvict = recordsCount - max - 1
+    return records[0..lastToEvict]
+}
+
+pipeline {
+    agent any
+    parameters {
+        booleanParam(name: 'doUpload', defaultValue: false, description: 'Flag controlling execution of upload.')
+        booleanParam(name: 'doEvict', defaultValue: true, description: 'Flag controlling execution of eviction.')
```

Comment:
Once this code gets moved to where it belongs, I suspect these flags won't be necessary anymore. They were just really useful for local testing.